control/preimage-kusama/src/commands.rs

```diff
@@ -0,0 +1,279 @@
+    let asset_id = get_asset_id(chain_id, contract);
+
+    register_asset_with_metadata(asset_id, min_balance, name, symbol, decimals, is_sufficient)
+}
+
+pub fn register_ether(
+    ether_min_balance: u128,
+    ether_name: String,
+    ether_symbol: String,
+    ether_decimals: u8,
+    ether_sufficient: bool,
+) -> (AssetHubRuntimeCall, AssetHubRuntimeCall) {
+    let chain_id = crate::constants::CHAIN_ID;
+    let asset_id = get_ether_id(chain_id);
+
+    register_asset_with_metadata(
+        asset_id,
+        ether_min_balance,
+        ether_name,
+        ether_symbol,
+        ether_decimals,
+        ether_sufficient,
+    )
+}
+
+pub fn register_asset_with_metadata(
+    asset_id: Location,
+    min_balance: u128,
+    name: String,
+    symbol: String,
+    decimals: u8,
+    is_sufficient: bool,
+) -> (AssetHubRuntimeCall, AssetHubRuntimeCall) {
+    let owner = xcm_helper::get_pah_owner_on_kusama();
+
+    let force_register =
+        AssetHubRuntimeCall::ForeignAssets(pallet_assets::pallet::Call2::force_create {
+            id: asset_id.clone(),
+            min_balance,
+            is_sufficient,
+            owner: MultiAddress::<AccountId32, ()>::Id(owner.into()),
+        });
+    let metadata =
+        AssetHubRuntimeCall::ForeignAssets(pallet_assets::pallet::Call2::force_set_metadata {
+            id: asset_id,
+            name: name.as_bytes().to_vec(),
+            symbol: symbol.as_bytes().to_vec(),
+            decimals,
+            is_frozen: false,
+        });
+
+    return (force_register, metadata);
+}
+
+pub fn token_registrations() -> Vec<AssetHubRuntimeCall> {
+    let mut calls = Vec::new();
+
+    // Ether
+    let (ether_create, ether_metadata) = register_ether(
+        15_000_000_000_000,
+        "Ether".to_string(),
+        "ETH".to_string(),
+        18,
+        true,
+    );
+    calls.push(ether_create);
+    calls.push(ether_metadata);
+
+    // Wrapped Ether (WETH)
+    let (weth_create, weth_metadata) = register_erc20_with_metadata(
+        hex!("c02aaa39b223fe8d0a0e5c4f27ead9083c756cc2"),
+        15_000_000_000_000,
+        "Wrapped Ether".to_string(),
+        "WETH".to_string(),
+        18,
+        true,
+    );
+    calls.push(weth_create);
+    calls.push(weth_metadata);
+
+    // USDT
+    let (usdt_create, usdt_metadata) = register_erc20_with_metadata(
+        hex!("dac17f958d2ee523a2206206994597c13d831ec7"),
+        10_000,
+        "USDT (Snowbridge)".to_string(),
+        "USDT".to_string(),
+        6,
+        true,
+    );
+    calls.push(usdt_create);
+    calls.push(usdt_metadata);
+
+    // USDC
+    let (usdc_create, usdc_metadata) = register_erc20_with_metadata(
+        hex!("a0b86991c6218b36c1d19d4a2e9eb0ce3606eb48"),
+        10_000,
+        "USDC (Snowbridge)".to_string(),
+        "USDC".to_string(),
+        6,
+        true,
+    );
+
+    calls.push(usdc_create);
+    calls.push(usdc_metadata);
+
+    // Staked USDe
+    calls.push(register_erc20(
```

Comment:
What about the asset metadata for these tokens?

Reply:
We do not have asset metadata for these tokens on Asset Hub Polkadot. Only sufficient assets need metadata set. Our advice to dapps was to always use the metadata from the Ethereum side.

Reply:
We don't have the asset metadata for these tokens on Polkadot. But I can add them for Kusama? WDYT?

Reply:
I rather match what we have in Polkadot Asset Hub.